event_model/__init__.py

```diff
@@ -442,6 +442,61 @@ def __init__(self, handler_registry, *,
         self.retry_intervals = retry_intervals
         self._closed = False
 
+    def __eq__(self, other):
+        return (
+            type(self) is type(other) and
+            self.inplace == other.inplace and
+            self._coerce == other._coerce and
+            self.include == other.include and
+            self.exclude == other.exclude and
+            self.root_map == other.root_map and
+            type(self._handler_cache) is type(other._handler_cache) and
+            type(self._resource_cache) is type(other._resource_cache) and
+            type(self._datum_cache) is type(other._datum_cache) and
+            type(self._descriptor_cache) is type(other._descriptor_cache) and
+            self.retry_intervals == other.retry_intervals
```

Comment:
Would it be unreasonable to compare `self.__getstate__()` and `other.__getstate__()` instead, to avoid the two implementations going out of sync? 

There is a slight difference, in that you're only checking type and not content of the caches, but as you pointed out in Slack that difference is a bit arbitrary.

Self-counter-argument: The test would become somewhat circular.
Counter-self-counter-argument: The test implies this should be guaranteed anyway.

Reply:
Yeah, I agree there's an argument for taking path, but I do think that treating the cache content as an internal is better.

Reply:
Makes sense; everything looks good to me.